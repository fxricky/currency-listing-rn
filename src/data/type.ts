export type Currency = {
  id: string;
  name: string;
  symbol: string;
  code?: string;
}
