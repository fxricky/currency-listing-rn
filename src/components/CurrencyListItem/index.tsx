import React from "react";
import { icChevronRight } from "@/assets/icons";
import { Currency } from "@/data/type";
import colors from "@/themes/colors";
import { Image, StyleSheet, Text, View } from "react-native";

type Props = {
  data: Currency;
};

function Avatar({ char }: { char: string }): React.ReactElement {
  return (
    <View style={styles.avatar}>
      <Text style={styles.avatarText}>{char}</Text>
    </View>
  );
}

export default function CurrencyListItem({ data }: Props): React.ReactElement {
  return (
    <View style={styles.container}>
      <Avatar char={data.name[0]} />
      <View style={styles.contentContainer}>
        <Text>{data.name}</Text>
        <View style={styles.row}>
          <Text>{data.symbol}</Text>
          <Image source={icChevronRight} style={styles.imgChevronRight} />
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    paddingHorizontal: 16,
    paddingVertical: 12,
    gap: 8,
  },
  contentContainer: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
  },
  row: {
    flexDirection: "row",
    alignItems: "center",
    gap: 8,
  },
  imgChevronRight: {
    width: 24,
    height: 24,
    tintColor: colors.gray["05"],
  },
  avatar: {
    width: 32,
    height: 32,
    borderRadius: 16,
    backgroundColor: colors.gray["07"],
    alignItems: "center",
    justifyContent: "center",
  },
  avatarText: {
    fontSize: 16,
    fontWeight: "bold",
    color: colors.gray["02"],
  },
});
