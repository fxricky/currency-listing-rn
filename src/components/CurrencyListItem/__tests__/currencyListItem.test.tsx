import { render } from "@testing-library/react-native";
import CurrencyListItem from "../index";
import { Currency } from "@/data/type";

const dummyData: Currency = {
  id: "1",
  name: "Bitcoin",
  symbol: "BTC",
};

describe("CurrencyListItem", () => {
  test("CurrencyListItem", () => {
    const tree = render(<CurrencyListItem data={dummyData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
