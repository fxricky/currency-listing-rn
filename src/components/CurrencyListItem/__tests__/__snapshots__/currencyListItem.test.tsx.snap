// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyListItem CurrencyListItem 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "gap": 8,
      "justifyContent": "space-between",
      "paddingHorizontal": 16,
      "paddingVertical": 12,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#364153",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      }
    }
  >
    <Text
      style={
        {
          "color": "#f3f4f6",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      B
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text>
      Bitcoin
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
        }
      }
    >
      <Text>
        BTC
      </Text>
      <Image
        source={
          {
            "testUri": "../../../src/assets/icons/icChevronRight.png",
          }
        }
        style={
          {
            "height": 24,
            "tintColor": "#99a1af",
            "width": 24,
          }
        }
      />
    </View>
  </View>
</View>
`;
